client/packages/common/src/ui/components/inputs/DatePickers/DateTimePickerInput/DateTimePickerInput.tsx

```diff
@@ -55,10 +89,25 @@ export const DateTimePickerInput: FC<
           },
         },
         textField: {
-          sx: { width: 250 },
+          error: !isInitialEntry && (!!error || !!internalError),
+          helperText: !isInitialEntry ? error ?? internalError ?? '' : '',
+          onBlur: e => {
+            onChange(DateUtils.getDateOrNull(e.target.value, format));
```

Comment:
The aforementioned `onBlur` to run `onChange` after keyboard entry. The catch here is that it passes the literal string input (the picker events above pass the constructed Date object), which is not in ISO format, hence I added that extension to `getDateOrNull` to parse other date string formats.